File: cpp/sorting/bubble_sort.cpp
void bubble_sort(int arr[], int n)
{
    for (int i = 0; i < n; i++)
    {
        bool is_swap = false;

        for (int j = 1; j < n - i; j++)
        {
            if (arr[j - 1] > arr[j])
            {
                int temp = arr[j];
                arr[j] = arr[j - 1];
                arr[j - 1] = temp;

                is_swap = true;
            }
        }

        if (!is_swap)
            break;
    }
}

File: sorting/bubble-sort.ts
function bubleSort(nums: number[]): number[] {
  let isSwap = 0;
  for (let i = 0; i < nums.length - 1; i++) {
    isSwap = 0;
    console.log(i);

    for (let j = i + 1; j <= nums.length - i; j++) {
      if (nums[j - 1] > nums[j]) {
        let temp = nums[j - 1];
        nums[j - 1] = nums[j];
        nums[j] = temp;

        isSwap++;
      }
    }
    if (isSwap == 0) break;
  }

  return nums;
}

console.log(bubleSort([1, 3, 9, 6]));
